Profiler/CProfilerCallback.h

```diff
@@ -160,4 +159,13 @@ class CProfilerCallback : public CProfilerCallbackBase {
 
 	/** Writes the fileVersionInfo into the provided buffer. */
 	int writeFileVersionInfo(LPCWSTR moduleFileName, char* buffer, size_t bufferSize);
+
+	HRESULT JITCompilationFinishedImplementation(FunctionID functionID, HRESULT hrStatus, BOOL fIsSafeToBlock);
+	HRESULT ShutdownImplementation();
+	HRESULT AssemblyLoadFinishedImplementation(AssemblyID assemblyID, HRESULT hrStatus);
+	HRESULT JITInliningImplementation(FunctionID callerID, FunctionID calleeID, BOOL *pfShouldInline);
+	HRESULT InitializeImplementation(IUnknown *pICorProfilerInfoUnk);
+
+	/** Logs a stack trace. If true is returned, the calling code should continue, otherwise it should rethrow the exception. */
+	bool handleException(std::string context);

```

Comment:
Why not rethrow the exception in this function?

Reply:
unfortunately, that's not possible. I have to catch (...) in the outer function so I can catch Windows structured exceptions as well (e.g. division by 0, general protection fault, ...) since these are not represented by actual exception objects. Thus, I can only rethrow via a throw; statement. And this only works in the context of a catch statement, which I don't have here anymore :-(